### PR TITLE
[Core] Add deprecation to StateMachineExtension

### PR DESCRIPTION
**Pull request by Wojdylak**
| Q               | A
|-----------------|-----
| Branch?         | 1.14
| Bug fix?        | no
| New feature?    | no
| BC breaks?      | no
| Deprecations?   | yes
| Related tickets | N/A
| License         | MIT

### CODE REVIEW

**Review by mpysiak — DISMISSED** [review]

File: src/Sylius/Bundle/CoreBundle/Twig/StateMachineExtension.php [hunk 1]
```diff
@@ -17,6 +17,7 @@
 use Twig\Extension\AbstractExtension;
 use Twig\TwigFunction;
 
+/** @deprecated since Sylius 1.14 and will be removed in Sylius 2.0. Use {@see \Sylius\Abstraction\StateMachine\Twig\StateMachineExtension} instead. */
```
Comment (GSadee) [inline src/Sylius/Bundle/CoreBundle/Twig/StateMachineExtension.php]: Use `trigger_deprecation` here

File: UPGRADE-1.14.md [hunk 2]
```diff
@@ -0,0 +1,3 @@
+# UPGRADE FROM `v1.13.X` TO `v1.14.0`
+
+1. The class `Sylius\Bundle\CoreBundle\Twig\StateMachineExtension` has been deprecated and will be removed in Sylius 2.0.
```
Comment (GSadee) [inline UPGRADE-1.14.md]: Add here also a note about replacement

File: UPGRADE-1.14.md [hunk 3]
```diff
@@ -0,0 +1,3 @@
+# UPGRADE FROM `v1.13.X` TO `v1.14.0`
+
+1. The class `Sylius\Bundle\CoreBundle\Twig\StateMachineExtension` has been deprecated and will be removed in Sylius 2.0. Use `Sylius\Abstraction\StateMachine\Twig\StateMachineExtension` instead.
```
Comment (NoResponseMate) [inline UPGRADE-1.14.md:3]: Noting the twig functions affected by this would be nice

Comment (Wojdylak) [reply UPGRADE-1.14.md:3]: The `Sylius\Bundle\CoreBundle\Twig\StateMachineExtension` has never been registered.

**Review by NoResponseMate — APPROVED** [review]

**Comment by NoResponseMate** [comment]
Thank you, @Wojdylak!

**Pull request merged**